webserver/webserver/src/main/java/io/helidon/webserver/http/HttpEntryPointsImpl.java

```diff
@@ -0,0 +1,151 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver.http;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.helidon.common.Weighted;
+import io.helidon.common.Weights;
+import io.helidon.common.types.Annotation;
+import io.helidon.common.types.TypedElementInfo;
+import io.helidon.service.registry.EntryPointInterceptor;
+import io.helidon.service.registry.InterceptionContext;
+import io.helidon.service.registry.Qualifier;
+import io.helidon.service.registry.Service;
+import io.helidon.service.registry.ServiceDescriptor;
+import io.helidon.service.registry.ServiceInstance;
+
+@Service.Singleton
+class HttpEntryPointsImpl implements HttpEntryPoint.EntryPoints {
+    private final boolean noInterceptors;
+    private final List<HttpEntryPoint.Interceptor> interceptors;
+
+    @Service.Inject
+    HttpEntryPointsImpl(List<ServiceInstance<EntryPointInterceptor>> entryPointInterceptors,
+                        List<ServiceInstance<HttpEntryPoint.Interceptor>> httpEntryPointInterceptors) {
+        this.noInterceptors = entryPointInterceptors.isEmpty() && httpEntryPointInterceptors.isEmpty();
+        this.interceptors = merge(entryPointInterceptors, httpEntryPointInterceptors);
+    }
+
+    @Override
+    public Handler handler(ServiceDescriptor<?> descriptor,
+                           Set<Qualifier> typeQualifiers,
+                           List<Annotation> typeAnnotations,
+                           TypedElementInfo methodInfo,
+                           Handler actualHandler) {
+
+        if (noInterceptors) {
+            return actualHandler;
+        }
+
+        InterceptionContext ctx = InterceptionContext.builder()
+                .typeAnnotations(typeAnnotations)
+                .elementInfo(methodInfo)
+                .serviceInfo(descriptor)
+                .build();
+
+        return new EntryPointHandler(ctx, interceptors, actualHandler);
+    }
+
+    private static List<HttpEntryPoint.Interceptor> merge(List<ServiceInstance<EntryPointInterceptor>> entryPoints,
+                                                          List<ServiceInstance<HttpEntryPoint.Interceptor>> httpEntryPoints) {
+
+        List<WeightedInterceptor> merged = new ArrayList<>();
+        httpEntryPoints.stream()
+                .map(it -> new WeightedInterceptor(it.get(),
+                                                   it.weight()))
+                .forEach(merged::add);
```

Comment:
IIRC ` .collect(Collectors.toList());` also creates new ArrayList 
```suggestion
        List<WeightedInterceptor> merged = 
                httpEntryPoints.stream()
                        .map(it -> new WeightedInterceptor(it.get(),
                                                           it.weight()))
                        .collect(Collectors.toList());
```

Reply:
`Collectors.toList())` is not ideal - it is way slower if I remember it correctly. Also  there was some uncertainity about its behavior.

Reply:
So I do not like the proposed approach - I would rather have a single list and add from both streams, then to use one stream to create the list and add fromd the second.
Concatenating the streams was too unreadable as well.